mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,263 @@
+            .format(repo=self.repo, id=issue_num)
+
+        response = requests.post(issue_labels_url, json.dumps(labels), auth=self.auth)
+        if response.status_code == 200:
+            logging.info('Successfully added labels to {}: {}.'.format(str(issue_num), str(labels)))
+            return True
+        else:
+            logging.error('Could not add the labels to {}: {}. \nResponse: {}'
+                          .format(str(issue_num), str(labels), json.dumps(response.json())))
+            return False
+
+    def remove_labels(self, issue_num, labels):
+        """
+        This method is to remove a list of labels to one issue.
+        It checks whether labels exist in the repo, and removes existing labels to the issue
+        :param issue_num: The specific issue number we want to label
+        :param labels: The labels which we want to remove
+        :return Response denoting success or failure for logging purposes
+        """
+        labels = self._format_labels(labels)
+        issue_labels_url = 'https://api.github.com/repos/{repo}/issues/{id}/labels/' \
+            .format(repo=self.repo, id=issue_num)
+
+        for label in labels:
+            delete_label_url = issue_labels_url + label
+            response = requests.delete(delete_label_url, auth=self.auth)
+            if response.status_code == 200:
+                logging.info('Successfully removed label to {}: {}.'.format(str(issue_num), str(label)))
+            else:
+                logging.error('Could not remove the label to {}: {}. \nResponse: {}'
+                              .format(str(issue_num), str(label), json.dumps(response.json())))
+                return False
+        return True
+
+    def update_labels(self, issue_num, labels):
+        """
+        This method is to update a list of labels to one issue.
+        It checks whether labels exist in the repo, and updates existing labels to the issue
+        :param issue_num: The specific issue number we want to label
+        :param labels: The labels which we want to remove
+        :return Response denoting success or failure for logging purposes
+        """
+        labels = self._format_labels(labels)
+        issue_labels_url = 'https://api.github.com/repos/{repo}/issues/{id}/labels' \
+            .format(repo=self.repo, id=issue_num)
+
+        response = requests.put(issue_labels_url, data=json.dumps(labels), auth=self.auth)
+        if response.status_code == 200:
+            logging.info('Successfully updated labels to {}: {}.'.format(str(issue_num), str(labels)))
+            return True
+        else:
+            logging.error('Could not update the labels to {}: {}. \nResponse: {}'
+                          .format(str(issue_num), str(labels), json.dumps(response.json())))
+            return False
+
+    def create_comment(self, issue_num, message):
+        """
+        This method will trigger a comment to an issue by the label bot
```

Comment:
Will this be done by the bot user or the committer account? I'd prefer the former to avoid confusion

Reply:
The question is which credentials it's going to use. If it uses the default credentials, it will post it under my name